src/app/code/community/Zendesk/Zendesk/Model/Observer.php

```diff
@@ -97,6 +97,20 @@ public function saveConfig(Varien_Event_Observer $observer)
                 Mage::getModel('core/config')->saveConfig('contacts/email/recipient_email', $oldEmail, $scope, $scopeId);
             }
         }
+
+        // If the zendesk domain is not found in the web widget snippet (wrapped with quotes), generate it again
+        $zDomain = Mage::getStoreConfig('zendesk/general/domain', $storeCode);
+        $widgetSnippet = Mage::getStoreConfig('zendesk/frontend_features/web_widget_code_snippet', $storeCode);
+        // Case insensitive search with single and double quotes, still better performance than 1 regexp search
+        if(stripos($widgetSnippet, "'{$zDomain}'") === false && stripos($widgetSnippet, '"'.$zDomain.'"') === false) {
+            $webWidgetSnippet=<<<EOJS
+<!-- Start of Zendesk Widget script -->
```

Comment:
This is the PHP [heredoc syntax](http://php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc) to work with long strings without needing to do concatenation, etc. Way cleaner and still allows you to interpolate variables as in line 108.